analyzer/runtime/extract.go

```diff
@@ -776,33 +774,33 @@ func extractEvents(blockData *BlockData, relatedAccountAddresses map[apiTypes.Ad
 						{
 							Name:    "from",
```

Comment:
maybe, but you'll have to make the argument in a much more fiddly way.

> the raw EVM event

"rawness" actually gets us away from any prescribed names. and when you go all the way down to the EVM layer, there are rather no events. at that layer, they're logs, with no data types or names.

a step above that is something about the ABI of these tokens, where there's the event name and types, but no argument names.

and then somewhere else, there's the ERC document itself, which has code snippets of some Solidity code for implementing the ABI, which uses these underscore-prefixed _from etc. names,

the actual verified contracts can call them whatever they want though, so there's no correct answer of what to put here in order to make things compatible.

but yeah, going forward I think it is the goal to produce these automatically from the ABI, using whatever names are there

Reply:
We're aligned overall. My terminology might have been sloppy: Where I said "raw EVM event", I meant "the EVM log as interpreted via the ABI", and "raw" only in the sense of "not restructured or renamed in any way by us".

It's a good point that implementers of an interface (say ERC-20) don't need to use the same function param names; I missed that. I agree we should ideally show whatever names the ABI uses.
Do you prefer to implement that here or in a separate PR? It seems it would be relatively easy here if callbacks like `ERC20Transfer` accepted the output of `abiParse.ParseEvent()` as their input, rather than the more tailored args like `fromECAddr` and `toECAddr`. cc @Andrew7234 whichever way you go because he'll need to implement basically the same stuff in his retroctive-parsing analyzer(s).

Reply:
mmmmmmm let's do it in a separate PR if we want to